### ISSUE TITLE
Update Vector's defaults to holistically solve durability

### ISSUE THREAD

**Issue by binarylogic** [vectordotdev/vector#10281]
This is a meta issue to look at Vector's durability defaults at a high level given some of the recent changes and improvements to buffers and end-to-end acknowledgements.

We attempted to solve this in the [Durability Brief](https://docs.google.com/document/d/19lW76R94Rf9AZ5lfZZyeXqa35xNQincc2a-Fx2xmKcI/edit?usp=sharing), but I don't think we landed on the right combination of changes given the in-flight work. Now that we're close to completing that work, we should reassess our defaults in a holistic manner to achieve durability by default.

## Proposal

(needs more discussion and consensus, many of the below issues overlap and a final design needs to be discussed)

- [ ] #10273
- [ ] #10280
- [ ] #3829
- [ ] #10282
- [ ] #10307
- [ ] #10326
- [ ] #10327
- [ ] #10392
- [ ] #10384
- [ ] #10381
- [ ] #10322

## Open questions & comments

1. If a downstream service goes down, Vector will retry sending data indefinitely in order to not lose data when/if the service comes back online. This poses a couple of problems:
   1. If the data came from the memory buffer, and end-to-end acknowledgements are enabled, the upstream client will eventually timeout and retry sending that data, creating duplicate data (this is not terrible).
   2. For business continuity, some users will want to failover to a backup service instead of retrying indefinitely.
2. Can we simplify this if consider the defaults in the context of push and pull-based sources. For example, push-based sources benefit from disk buffers while pull-based do not since they typically handle buffering and durability.

**Comment by scMarkus** [comment]
Is there any progress going on with this rework? I would be greatly interested in trading memory buffers for disk buffers (#10392).

**Comment by bruceg** [comment]
No, no progress on most of this. Some of it is planned but deprioritized.

**Comment by scMarkus** [comment]
Is there a particular part I could help with? I am quite interested in this progressing and would be willing to actively support it.

**Comment by scMarkus** [comment]
Hi,
is there any progress on this one? I would still be very much interested in seeing this progressing. Also I would be happy to actively assist.
